libvast/include/vast/table_slice.hpp

```diff
@@ -314,14 +283,39 @@ class table_slice final {
 
 // -- operations ---------------------------------------------------------------
 
+/// Joins all slices in the given range.
+/// @param slices The input table slices.
+table_slice join(std::vector<table_slice> slices);
```

Comment:
The name is confusing. When I read "join", I am thinking "what's the key you're joining on?". If you're literally concatenating them, I have some suggestions:

1. `concat(enate)`
2. `bind_rows` (dplyr style, which also has `bind_cols`)
3. `stack`

But the vertical data fusion must come up, which is definitely not the case with `join`.

Reply:
I like `concat`, and definitely agree on `join` being confusing since we're operating on two-dimensional data here. Thanks for the suggestions!

Reply:
I renamed it to `concatenate`.